src/utils/network-monitor.ts

```diff
@@ -1517,49 +1519,52 @@ export class NetworkMonitor {
     canFail = false,
     interval = 10_000,
   }: LogsParams): Promise<Log[] | null> {
-    return new Promise<Log[] | null>((topResolve, _topReject) => {
-      let counter = 0
-      let sent = false
-      let logsInterval: NodeJS.Timeout | null = null
-      const targetBlock: string = BigNumber.from(blockNumber).toHexString()
-      const getLogs = async (): Promise<void> => {
-        try {
-          const filter: Filter = {fromBlock: targetBlock, toBlock: targetBlock}
-          const logs: Log[] | null = await this.providers[network].getLogs(filter)
-          if (logs === null) {
-            counter++
-            if (canFail && counter > attempts) {
-              if (logsInterval) clearInterval(logsInterval)
-              if (!sent) {
-                sent = true
-                topResolve(null)
-              }
-            }
-          } else {
-            if (logsInterval) clearInterval(logsInterval)
+    let counter = 0
+    let sent = false
+    const targetBlock: string = BigNumber.from(blockNumber).toHexString()
+
+    // eslint-disable-next-line no-unmodified-loop-condition
```

Comment:
I tried to avoid the need for this override, but `canFail` is passed through as a parent function parameter so it can either be false or true and doesn't change for the duration of the function so it might require a bigger refactor to remove this functionality or move it into another block. Not sure yet. open to ideas, but mainly just want to simplify things and have it working at the moment

Reply:
It would be good to see where canFail is set to true or falseZ that way we would know if we have to support it.

Reply:
it's used with canFail set to false for a few of the functions and the network monitor also has instances where canFail is set to true so we do need to support it to maintain current functionality

Reply:
see `getLogs` `getBlock` etc